interpreter/binary/decode.ml

```diff
@@ -499,12 +509,39 @@ let rec instr s =
 and instr_block s = List.rev (instr_block' s [])
 and instr_block' s es =
   match peek s with
-  | None | Some (0x05 | 0x0b) -> es
+  | None | Some (0x05 | 0x07 | 0x0a | 0x0b | 0x18 | 0x19) -> es
   | _ ->
     let pos = pos s in
     let e' = instr s in
     instr_block' s (Source.(e' @@ region s pos pos) :: es)
 
+and handler s =
+  let rec catch_list () =
```

Comment:
Nit: avoid nesting functions unless they close over local variables in a non-trivial fashion.

Reply:
Done.